Kubectl set-context fails because .kube/config file is read-only

I am having a problem with starting kraken in our OpenShift 4.6 cluster:

error: open /root/.kube/config.lock: read-only file system
2021-07-26 13:58:48,953 [ERROR] Failed to run kubectl config set-context --current --namespace=openshift-kraken, error: Command 'kubectl config set-context --current --namespace=openshift-kraken' returned non-zero exit status 1.

The problem is according to your guide the .kube/config file is mounted via configmap to the correct folder. If you now execute a context switch in the code, kubectl tries to write this to the .kube/config file which is not possible as configmaps can only be mounted as read-only into containers. Do you have a solution for that?

--- a/kraken/kubernetes/client.py
+++ b/kraken/kubernetes/client.py
@@ -245,13 +245,13 @@ def find_kraken_node():
     if kraken_pod_name:
         # get kraken-deployment pod, find node name
         try:
-            runcommand.invoke("kubectl config set-context --current --namespace=" + str(kraken_project))
             node_name = runcommand.invoke(
-                "kubectl get pods/" + str(kraken_pod_name) + ' -o jsonpath="{.spec.nodeName}"'
+                "kubectl get pods/"
+                + str(kraken_pod_name)
+                + ' -o jsonpath="{.spec.nodeName}"'
+                + " -n"
+                + str(kraken_project)
             )
-
-            # Reset to the default project
-            runcommand.invoke("kubectl config set-context --current --namespace=default")
 
             global kraken_node_name
             kraken_node_name = node_name

